packages/lit-element/src/lib/lit-element.ts

```diff
@@ -126,83 +140,47 @@ export class LitElement extends UpdatingElement {
    * using the [[`css`]] tag function or via constructible stylesheets.
    */
   static styles?: CSSResultOrNative | CSSResultArray;
-
-  private static _styles: Array<CSSResultOrNative | CSSResult> | undefined;
+  private static _elementStyles?: CSSResultFlatArray;
 
   /**
-   * Return the array of styles to apply to the element.
+   * Takes the styles the user supplied via the `static styles` property and
+   * returns the array of styles to apply to the element.
    * Override this method to integrate into a style management system.
    *
+   * Styles are deduplicated preserving the _last_ instance in the list. This
+   * is a performance optimization to avoid duplicated styles that can occur
+   * especially when composing via subclassing. The last item is kept to try
+   * to preserve the cascade order with the assumption that it's most important
+   * that last added styles override previous styles.
+   *
    * @nocollapse
    */
-  static getStyles(): CSSResultOrNative | CSSResultArray | undefined {
-    return this.styles;
-  }
-
-  /** @nocollapse */
-  private static _getUniqueStyles() {
-    // Only gather styles once per class
-    if (this.hasOwnProperty(JSCompiler_renameProperty('_styles', this))) {
-      return;
+  protected static getStyles(
```

Comment:
Since this is a braking change to the API anyway, I think a name change would be good.  This is is now a transformer, not a getter, in that it works only on the argument passed in, not on any state.  `flattenStyles`? `transformStyles`?

Reply:
Renamed to `finalizeStyles`.